src/m365/spo/commands/commandset/commandset-add.ts

```diff
@@ -0,0 +1,156 @@
+import { Logger } from '../../../../cli/Logger';
+import GlobalOptions from '../../../../GlobalOptions';
+import commands from '../../commands';
+import { validation } from '../../../../utils/validation';
+import SpoCommand from '../../../base/SpoCommand';
+import request from '../../../../request';
+import { CustomAction } from '../customaction/customaction';
+
+interface CommandArgs {
+  options: Options;
+}
+
+export interface Options extends GlobalOptions {
+  title: string;
+  webUrl: string;
+  listType: string;
+  clientSideComponentId: string;
+  clientSideComponentProperties?: string;
+  scope?: string;
+  location?: string;
+}
+
+class SpoCommandsetAddCommand extends SpoCommand {
+  private static readonly listTypes: string[] = ['List', 'Library'];
```

Comment:
sorry for the late idea (we may totally ignore it for now as it was not part of the issue and add it later) but I was thinking if maybe we could also add SitePages here 🤔. I think it is also quite common case for adding a command set button and still it will not be covered by the 101 (library). 
If I am not mistaken the Site Pages list template is 119.
@martinlingstuyl since you are the master mind behind this command I am also interested in your opinion 😉.
and any other from @pnp/cli-for-microsoft-365-maintainers 👍.
Once again sorry for being late to the party and I am totally fine with ignoring it for now, merge as is, and add it later on 👍

Reply:
No problem! I like the idea that you're thinking further.

Let's wait for Martin or other maintainers to give their opinion about this. No rush here

Reply:
@martinlingstuyl did you find some time to have a check on this one ?

Reply:
I think it's a good idea to add other common targets to add commandsets to. In favor of including Site Pages in the list.

Reply:
@nicodecleyre may I kindly ask you to implement the change?

Reply:
@waldekmastykarz thanks for your feedback 👍

Reply:
> @nicodecleyre may I kindly ask you to implement the change?

You sure can 😉 will happely implement the change